src/main/java/it/pagopa/ecommerce/commons/utils/JwtTokenUtils.java

```diff
@@ -0,0 +1,81 @@
+package it.pagopa.ecommerce.commons.utils;
+
+import io.jsonwebtoken.JwtBuilder;
+import io.jsonwebtoken.JwtException;
+import io.jsonwebtoken.Jwts;
+import it.pagopa.ecommerce.commons.domain.Claims;
+import it.pagopa.ecommerce.commons.exceptions.JWTTokenGenerationException;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Component;
+import reactor.core.publisher.Mono;
+import javax.crypto.SecretKey;
+import javax.validation.constraints.NotNull;
+import java.time.Duration;
+import java.time.Instant;
+import java.util.Date;
+import java.util.UUID;
+
+/**
+ * This class generate jwt token
+ */
+@Component
+@Slf4j
+public class JwtTokenUtils {
+
+    /**
+     * The claim transactionId
+     */
+    public static final String TRANSACTION_ID_CLAIM = "transactionId";
+
+    /**
+     * The claim orderId
+     */
+    public static final String ORDER_ID_CLAIM = "orderId";
+
+    /**
+     * The claim payment methodId
+     */
+    public static final String PAYMENT_METHOD_ID_CLAIM = "paymentMethodId";
+
+    /**
+     * This method generates a jwt with specific claim
+     *
+     * @param jwtSecretKey             jwt secret key
+     * @param tokenValidityTimeSeconds jwt validity time
+     * @param claims                   value object with all claims to set into jwt
+     * @return Mono jwt with specific claim
+     */
+    public Mono<String> generateToken(
```

Comment:
Maybe this method can directly return jwt token instead of a Mono since no async operation is performed here. What do you think?

Reply:
I think it is better to do it in a dedicated PR as now the refactoring would have an impact on the times 🙂

Reply:
Fixed with ae80c88d29b36f3ac7a5b17231e864a671d47a6b! What do you think?